locator/abstract_replication_strategy.cc

```diff
@@ -165,19 +165,35 @@ effective_replication_map::get_ranges(inet_address ep) const {
     });
 }
 
-// Caller must ensure that token_metadata will not change throughout the call if can_yield::yes.
+// Caller must ensure that token_metadata will not change throughout the call.
 future<dht::token_range_vector>
 abstract_replication_strategy::get_ranges(inet_address ep, token_metadata_ptr tmptr) const {
```

Comment:
nit: it's actually easy to get rid of this entry point since it's needed only by a couple places in repair that already capture the tmptr in a seastar thread function.

Reply:
Yes, but can be done later.